changelog.d/pr-7273.md

```diff
@@ -0,0 +1,3 @@
+### 🐛 Bug Fixes
+
+- create-sibling: Use C.UTF-8 locale instead of C.  [PR #7273](https://github.com/datalad/datalad/pull/7273) (by [@nobodyinperson](https://github.com/nobodyinperson))
```

Comment:
```suggestion
- create-sibling: Use C.UTF-8 locale instead of C on the remote end.  [PR #7273](https://github.com/datalad/datalad/pull/7273) (by [@nobodyinperson](https://github.com/nobodyinperson))
```